### PR TITLE
merge feat improvment end refactor into develop

### PR DESCRIPTION
**Pull request by BBR2394**
I done Patch for medicine.

### CODE REVIEW

File: app/crud/crud_medicine.py [hunk 1]
```diff
@@ -40,6 +40,42 @@ def add_one_med(self, new_medicine: medicine_model.medecineShort, dbSession: Ses
...
+        return elem_to_del
+
+    def join_get_type_of_a_medicine(self, dbSession: Session, id_med: int):
+        print("dans join_get_type_of_a_medicine")
+
+        resp = dbSession.query(medicine_model.medicine).get(id_med)
+        print(resp)
+        type_list = [dbSession.query(medicine_model.medecine_type).get(current_id) for current_id in resp.list_type]
+        # list_type_medicine = dbSession.query(medicine_model.medicine).filter(medicine_model.medicine.id == id_med).join(medicine_model.medecineType).filter(medicine_model.medecineType == medicine_model.medicine.list_type).all()
+        print(type_list)
```
Comment (Unknow46) [inline app/crud/crud_medicine.py]: Supprimer les print 😸

File: app/routers/medicine_routes.py [hunk 2]
```diff
@@ -45,12 +45,30 @@ def add_a_medicine(body_medicine: medicine_schema.medicinePost, dbSession: Sessi
...
     return new_med_created
 
-# TODO: path
+
 @medicine_router_bis.patch("/")
-def get_all_medicine(db: Session = Depends(db_dependencies.get_db), limit: int = 10):
-    return { "message": "in progress PATCH medicine"}
+def get_all_medicine(body_medicine: medicine_schema.medicinePatch, dbSession: Session = Depends(db_dependencies.get_db)):
+    print("dans PATCH medicine")
+    print(body_medicine)
```
Comment (Unknow46) [inline app/routers/medicine_routes.py]: Supprimer les print 😸

File: app/routers/medicine_routes.py [hunk 3]
```diff
@@ -45,12 +45,30 @@ def add_a_medicine(body_medicine: medicine_schema.medicinePost, dbSession: Sessi
...
+    print(body_medicine.id)
+    elem_deleted = crudMed.delete_medicine(dbSession, body_medicine.id)
+    #elem_deleted = crudBaseOBJMedicine.remove(dbSession, model_id=body_medicine.id)
+    return elem_deleted
+
+# cette fonction retourne une liste de string corespondant au type de medoc existant
+@medicine_router_bis.get("/type/{id_med}")
+def get_all_type_medicine(id_med: int, dbSession: Session = Depends(db_dependencies.get_db)):
+    print("In GET type according to ID")
+    print(id_med)
```
Comment (Unknow46) [inline app/routers/medicine_routes.py]: Supprimer les print 😸

**Review by Unknow46 — COMMENTED** [review]
Magnifique 😸

File: app/crud/crud_medicine.py (hunk 1 above)
Comment (BBR2394) [reply app/crud/crud_medicine.py]: normalement done

**Review by Unknow46 — APPROVED** [review]

**Pull request merged**